src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -869,5 +905,22 @@ void DefaultSceneTableImpl::SceneEntryIteratorImpl::Release()
     mProvider.mSceneEntryIterators.ReleaseObject(this);
 }
 
+namespace {
+
+static DefaultSceneTableImpl gSceneTableImpl;
+
+} // namespace
+
+/// @brief Instance getter for the default global scene table implementation
+///        Should only be used in the scene cluster
+/// @return Default global scene table implementation
+DefaultSceneTableImpl * GetSceneTableImpl(EndpointId endpoint, uint16_t endpointTableSize)
+{
+    gSceneTableImpl.SetEndpoint(endpoint);
```

Comment:
This could use some significant documentation.  Why is it ok to just change the endpoint of the gSceneTableImpl?  Is it because we expect people to never cache the returned pointer an always call GetSceneTableImpl() right before using it and it does not actually store any state in members apart from the endpoint and size?  And we never expect people to operate on multiple endpoints "at once"?  But this table _absolutely_ stores state in mAVPairs and mValueBuffer.  So what is actually going on here?

For now, please document this all very clearly in the API.  But we should think carefully about how to make this safer.  For example,

Reply:
The access to the SceneTable should be made thread safe. The scene table generates a RAM image of the table in flash upon use and the proper endpoint should be chosen upon each call. 
Not picking and endpoint will result in preventing flash usage.

mAVPairs and mValueBuffer were an oversight, a cleanup will need to be added on handlers.